### DISCUSSION TITLE
404 return code for API endpoint

### DISCUSSION THREAD

**Discussion by neo2271** [vercel/vercel#6540] [Help] [answered]
Hi, 

I have met an issue when adding a new version for API endpoints.

Below is my current API directory:
![image](https://user-images.githubusercontent.com/2898072/127768001-15559c80-15ce-4628-a690-299a872af49d.png)

Before I just used "api/v1/index.js". API endpoints works normally.

After that, I added a new version ("api/v2/"). Below is my "api/v2/index.js":

```
const express = require("express");
const app = express();

app.use(express.json());
app.use(express.urlencoded({ extended: true }));

app.get('/api/v2', async (req, res) => {
    // console.log(req.headers);
    let message = "Welcome to /api/v2/ endpoints...";
    return res.status(200).json({ message: message });
});

app.get('/api/v2/test', async (req, res) => {
    // console.log(req.headers);
    let message = "Hello, I'm a /api/v2/test";
    return res.status(200).json({ message: message });
});

app.get('/test', async (req, res) => {
    // console.log(req.headers);
    let message = "Hello, I'm a test routed from /api/v2/";
    return res.status(200).json({ message: message });
});

module.exports = app;
```

The "/api/v2" endpoint work nornally but **other endpoints** (in "/api/v2/index.js") **got the 404 return code**.
![image](https://i.imgur.com/8bcWfmO.png)
![image](https://i.imgur.com/xlHRwix.png)
![image](https://i.imgur.com/Z19PfEw.png)

The configuration from my vercel.json:

```
{
    "name": "serverless-minhtamgroup",
    "version": 2,
    "rewrites": [
    {
        "source": "/(.*)",
        "destination": "/api/v1"
    },
    {
        "source": "/api/v1/(.*)",
        "destination": "/api/v1"
    },
    {
        "source": "/(.*)",
        "destination": "/api/v2/index.js"
    },
    {
        "source": "/api/v2/(.*)",
        "destination": "/api/v2/index.js"
    }]
}
```

Thanks in advance!

**Comment by leerob** [comment]
I would recommend reading this article: https://vercel.com/guides/using-express-with-vercel

**Comment by neo2271** [comment]
I have followed that article to build my v1 APIs. But when I added v2, I have no idea why above issue happened

**Reply by leerob** [reply]
I would recommend using Vercel's functions by default instead of Express here 👍🏼

**Comment by neo2271** [comment, marked as answer]
I have moved all my APIs into an unique file /api/index.js. I have to manage endpoints (/api/v1 and /api/v2) in this file.
